src/briefcase/config.py

```diff
@@ -146,6 +146,44 @@ def validate_document_type_config(document_type_id, document_type):
             f"The URL associated with document type {document_type_id!r} is invalid: {e}"
         )
 
+    if sys.platform == "darwin":  # pragma: no-cover-if-not-macos
+        from briefcase.platforms.macOS.utils import mime_type_to_UTI
+
+        macOS = document_type.setdefault("macOS", {})
+        mime_type = document_type.get("mime_type", None)
+        if (uti := mime_type_to_UTI(mime_type)) is not None:
+            macOS.setdefault("is_core_type", True)
+            macOS.setdefault("LSItemContentTypes", uti)
+            macOS.setdefault("LSHandlerRank", "Alternate")
+        else:
+            # LSItemContentTypes will default to bundle.app_name.document_type_id
+            # in the Info.plist template if it is not provided.
+            macOS.setdefault("is_core_type", False)
+            macOS.setdefault("LSHandlerRank", "Owner")
+            macOS.setdefault("UTTypeConformsTo", ["public.data", "public.content"])
+
+        macOS.setdefault("CFBundleTypeRole", "Viewer")
+
+        content_types = macOS.get("LSItemContentTypes", None)
+        if isinstance(content_types, list):
+            if len(content_types) > 1:
+                raise BriefcaseConfigError(
+                    f"""
+Document type {document_type_id!r} has multiple content types. Specifying
+multiple values in a LSItemContentTypes key is only valid when multiple document
+types are manually grouped together in the Info.plist file. For Briefcase apps,
+document types are always separately declared in the configuration file, so only
+a single value should be provided.
+                """
+                )
+            else:
+                macOS["LSItemContentTypes"] = content_types[0]
```

Comment:
While I agree that there can only be one LSItemContentType entry, and we should be validating for this case,  I'd prefer to keep the pluralisation of LSItemContentTypes aligned with the storage - that is, normalise *to* a list, and then only use the `[0]` element in the template.

Reply:
Done!